File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        // nums.sort(())
        const set = new Set(nums);
        let longest = 0;

        for (let i = 0; i < nums.length; i++) {
            let streak = 0;
            let curr = nums[i];
            while (set.has(curr++)) {
                streak++;
            }
            longest = Math.max(longest, streak);
        }

        return longest;
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        cache = {}
        def dfs(i):
            if i >= len(nums):
                return 0

            if i in cache:
                return cache[i]
            val = nums[i] + max(dfs(i + 2), dfs(i + 3))
            cache[i] = val
            return val
        return max(dfs(0), dfs(1), dfs(2))
        